app/controllers/signup_controller.rb

```diff
@@ -3,20 +3,16 @@ class SignupController < ApplicationController
   PROFILE_TIMEOUT = 30.minutes
 
   skip_before_filter :authenticate_user!,
-                     only: [:start, :verify_email, :verify_by_token, :password, :social, :profile]
+                     only: [:start, :verify_email, :verify_by_token, :trusted_student, :password, :social, :profile, :trusted]
 
   skip_before_filter :complete_signup_profile
 
   fine_print_skip :general_terms_of_use, :privacy_policy
 
   before_filter :check_ready_for_profile, only: [:profile]
 
-  # TODO spec this
   before_filter :restart_if_missing_signup_state, only: [:verify_email, :password, :social]
-
-  # TODO spec this
-  before_filter :exit_signup_if_logged_in, only: [:start, :verify_email, :password, :social, :verify_by_token]
-
+  before_filter :exit_signup_if_logged_in, only: [:start, :social, :verify_by_token]
```

Comment:
I'm still concerned that `:verify_email`, `:password`, and `:social` are no longer in this `only` list... sorry if you explained this in a prior review response... do we know these are safe to remove?

Reply:
I don't recall why I removed these..  And I thought these were specced, but It now appears they are not since the spec didn't fail.   I'll add add them back and add a spec for 'em.